lib/spack/spack/main.py

```diff
@@ -563,15 +568,18 @@ def main(argv=None):
         parser.print_help()
         return 1
 
-    # -h and -V are special as they do not require a command, but all the
-    # other options do nothing without a command.
+    # -h, -H, and -V are special as they do not require a command, but
+    # all the other options do nothing without a command.
     if not args.command:
         if args.version:
```

Comment:
Was wondering if we could simplify a bit this and write:
```python
if args.version:
    ...
elif args.help:
    ...
elif not args.command:
    parser.print_help()
    return 1
```
and delete the lines that deal with `args.version` and `args.help` below.

Reply:
I think this was originally separated to prevent subcommand arguments with the same name as `spack` arguments from setting things at the `spack` level.  But the way the parser works now, where `COMMAND` is a dummy argument and we use `argparse.REMAINDER`, seems to handle that.  So I simplified it!  Yay.